### ISSUE TITLE
[SPRINT] Determine halo orientation from downsampled particles

### ISSUE THREAD

**Issue by jablazek** [LSSTDESC/SprintWeek2021#17]
# Determine halo orientation from downsampled particles

Test the impact of limited particle number in the precision of measured halo orientation.

Contacts: @jablazek @nvanalfen @EiffL 
Day/Time: All week
Main communication channel: slack, zoom: https://northeastern.zoom.us/j/8435767380
GitHub repo:

## Goals and deliverable
Using SkySim5000 or other data, we want to answer the following questions:
- What is the minimum number of DM particles in a halo required for a "reasonable" measurement of the halo orientation? Note that this is not the same as the full halo shape.
- How does the noise in the measured orientation angle scale with particle number?

## Resources and skills needed
Working with particle data to extract halo properties.

## Detailed description
- Start with a catalog of well-resolved halos. Measure their orientation angles. This is "truth."
- Downsample by progressively larger factors, until we are left with only 5-10 particles per halo. For each downsampling, we could try multiple realizations to further examine the scatter.
- Plot Theta as a function of particle number, where Theta is the angle between the measured axis and the true axes.

**Comment by aphearin** [comment]
In addition to the tests outlined above, we may also find it useful to study the resolution question by looking at idealized ellipsoidal NFW halos. It is easy to generate such halos with [this halotools function.](https://halotools.readthedocs.io/en/latest/api/halotools.empirical_models.NFWPhaseSpace.html#halotools.empirical_models.NFWPhaseSpace.mc_generate_nfw_phase_space_points)

The way this test would work is we would use the same shape-measuring code on a very well-sampled, idealized ellipsoidal halo, and see how the measured orientation jiggles as we downsample the ellipsoid. This test is complementary to the others outlined above because it zeros in on the true underlying noise, and so allows us to isolate the root cause of the stochasticity from its manifestation on the two-point function.

**Comment by aphearin** [comment]
In case we want to run tests on smaller simulations whose downsampled particles do not require multiple compute nodes, [this halotools function](https://halotools.readthedocs.io/en/latest/api/halotools.mock_observables.inertia_tensor_per_object.html) may also be useful. I'm guessing it provides the single-node analog of the same computation used to extract shapes from the HACC sims.

**Comment by evevkovacs** [comment]
@jablazek Will you need particle data from the ANL simulations or do you already have simulation data that would be suitable for this study?

**Comment by jablazek** [comment]
@evevkovacs : this would be the full particle data for the massive halos in SkySim5000, right?

@aphearin @EiffL : Should we start with a different set of particle data (perhaps going to lower masses), or the idealized ellipsoids Andrew suggested? If there is time, we should see how the downsampling impacts both the idealized and realistic cases.

**Comment by aphearin** [comment]
For purposes of getting the project off the ground quickly, it may be easier to begin working with a laptop-sized simulation that has publicly available particle data.

**Comment by evevkovacs** [comment]
@jablazek @aphearin  Yes, the question was about whether we would need a sample halo particle file from the Outer Rim simulation.  But Andrew has answered that at least to start with, no.  Then let me rephrase: Is it likely that, for the purposes of this sprint, you would need any of the full particle data from the Outer Rim simulation? Some of it is on tape, it is not stored halo by halo, and is not in a readily readable format, so we can't provide it for you immediately on request. I just wanted to alert you to the fact that there will be some overheads in acquiring and using these data.

**Comment by aphearin** [comment]
Seems to me that just using laptop-sized catalogs like the ones I provide with halotools would make it easier to make fast progress during sprint week. But I'm guessing that @jablazek and co., eventually want to work with a big sim.

**Comment by patricialarsen** [comment]
The conclusion I've come to is that processing the low mass halos might be possible, but it would be a significant amount of effort to take the relevant data products off tape and re-process them. 

When you're looking at this sprint, if you could make sure to check if any conclusions using idealized cases hold up for simulated, oddly shaped halos of the relevant mass resolution, what the level of benefit overall of getting a noisy measurement vs a random orientation as you go down in mass is, and whether the level of results you get from the noisy measurements are significantly improving the science case, that would be really useful to see in prioritizing this work.

**Comment by aphearin** [comment]
Here's some code that should be useful for generating triaxial NFW halo profiles https://github.com/aphearin/ellipsoidal_nfw
